packages/shared-ux/chrome/serverless_projects_documentation.md

```diff
@@ -0,0 +1,497 @@
+      <div>Your custom UI</div>
+    </Navigation.Item>
+  </Navigation.Group>
+</Navigation>
+```
+
+### Important Concepts
+
+#### Deep links
+
+[Deep links](https://github.com/elastic/kibana/blob/f5034e60a501e7b61a3e1bff34e64c9b94c71344/packages/core/application/core-application-browser/src/application.ts#L281-L314) are a Kibana core's mechanism for registering sub-pages within an app. In "classic" Kibana they are used for the default side navigation, navigation APIs and the global search. Teams can register the deep links when they register their app. They can also update or remove deep links dynamically.
+
+The serverless navigation API uses the same deep links mechanism to configure the navigation tree. The `link` property of the `NodeDefinition` interface refers to the deep links registered by apps. The `link` property can be either a direct app id or a deep link ID.
+
+There are multiple benefits of using deep links instead of the hardcoded URLs when configuring the navigation tree:
+
+- Validation: the deep links are validated when the tree is built. If the deep link wasn't registered by any of the apps, an error will be thrown.
+- Type safety: the list of deep links is typed, and we don't rely on static URLs that can break.
+- Dynamic updates: the deep links can be updated or removed dynamically by apps that own them. The navigation tree will be updated accordingly.
+
+Internal navigation should be configured using deep links. The `href` property should be used only for external links. There is also a special type of external links for links pointing to the cloud console - `cloudLink`.
+
+#### Cloud links
+
+The `cloudLink` property of the `NodeDefinition` interface refers to the predefined list of cloud links that are configured in `kibana.yml`.
+Currently available pages are `'userAndRoles' | 'performance' | 'billingAndSub'`
+
+```ts
+import { type NavigationTreeDefinition, getPresets } from '@kbn/shared-ux-chrome-navigation';
+
+const navigationTree: NavigationTreeDefinition = {
+  body: [
+    {
+      type: 'navGroup',
+      id: 'topNav',
+      children: [
+        {
+          id: 'cloudLinkUserAndRoles',
+          cloudLink: 'userAndRoles',
+        },
+        {
+          id: 'cloudLinkPerformance',
+          cloudLink: 'performance',
+        },
+      ],
+    },
+  ],
+};
+```
+
+#### Preconfigured navigation sections
+
+When configuring the navigation tree you can use the preconfigured sections like `devtools`, `management`, `ml`, `analytics`
+
+```ts
+import { type NavigationTreeDefinition, getPresets } from '@kbn/shared-ux-chrome-navigation';
+
+const navigationTree: NavigationTreeDefinition = {
+  body: [
+    {
+      type: 'navGroup',
+      ...getPresets('ml'), // Insert all the machine learning links
+    },
+  ],
+};
+```
+
+The benefit of using the preset instead of building the sectional manually is that the team who can own the preset can update it as they see fit.
```

Comment:
```suggestion
The benefit of using the preset instead of building the platform-owned sections manually is that the team who can own the preset can update it as they see fit.
```